pkg/ruler/manager.go

```diff
@@ -141,33 +140,53 @@ func (r *DefaultMultiTenantManager) syncRulesToManager(ctx context.Context, user
 		return
 	}
 
+	if !update {
```

Comment:
This is not the same logic we had before. We need to check if it exists first. Rules may be synced on disk, but the manager not exist in memory.